compiler/noirc_evaluator/src/ssa/opt/remove_if_else.rs

```diff
@@ -241,3 +241,148 @@ fn slice_capacity_change(
         | Intrinsic::FieldLessThan => SizeChange::None,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::ssa::{opt::assert_normalized_ssa_equals, ssa_gen::Ssa};
+
+    #[test]
+    fn merge_basic_arrays() {
+        // This is the flattened SSA for the following Noir logic:
+        // ```
+        // fn main(x: bool, mut y: [u32; 2]) {
+        //     if x {
+        //         y[0] = 1;
+        //         y[1] = 2;
+        //     }
+        //
+        //     let z = y[0] + y[1];
+        //     assert(z == 3);
+        // }
+        // ```
+        let src = "
+        acir(inline) predicate_pure fn main f0 {
+          b0(v0: u1, v1: [u32; 2]):
+            v2 = allocate -> &mut [u32; 2]
+            enable_side_effects v0
+            v5 = array_set v1, index u32 0, value u32 1
+            v7 = array_set v5, index u32 1, value u32 2
+            v8 = not v0
+            v9 = if v0 then v7 else (if v8) v1
+            enable_side_effects u1 1
+            v11 = array_get v9, index u32 0 -> u32
+            v12 = array_get v9, index u32 1 -> u32
+            v13 = add v11, v12
+            v15 = eq v13, u32 3
+            constrain v13 == u32 3
+            return
+        }
+        ";
+
+        let mut ssa = Ssa::from_str(src).unwrap();
+
+        ssa = ssa.remove_if_else();
+        println!("{}", ssa);
```

Comment:
Assuming this was a left-over to get the original SSA:

```suggestion
        ssa = ssa.remove_if_else();
```

Reply:
(but otherwise looks great, thank you!)